tuning/config/acceleration_configs/attention_and_distributed_packing.py

```diff
@@ -0,0 +1,35 @@
+# Standard
+from dataclasses import dataclass
+
+# Local
+from .utils import ensure_nested_dataclasses_initialized, parsable_dataclass
+
+
+@parsable_dataclass
+@dataclass
+class PaddingFree:
+
+    method: str = "huggingface"
+
+    def __post_init__(self):
+        if self.method != "huggingface":
+            raise ValueError("only 'huggingface' method currently supported.")
+
+
+@parsable_dataclass
+@dataclass
+class MultiPack:
+
+    num_processes: int = 16
```

Comment:
Is there any guidance on what this number should be set to?

Reply:
this number is a reasonable one for most datasets of reasonable size (e.g. under a million examples). The packing algorithm is relatively fast, but in the event the dataset is too large, then our plugin will raise a warning
https://github.com/foundation-model-stack/fms-acceleration/blob/4e81c64453ec5d2b06a8d14a2a72374cc736098a/plugins/attention-and-distributed-packing/src/fms_acceleration_aadp/framework_plugin_multipack.py#L117-L123

that advises the user to increase this number if  the process times out.